src/live_restore/live_restore_fs.c

```diff
@@ -171,6 +171,15 @@ __live_restore_fs_create_tombstone(
     lr_fs = (WTI_LIVE_RESTORE_FS *)fs;
     path = path_marker = NULL;
 
+    /*
+     * TODO: This is a big old race where we set this flag outside a lock and therefore have no idea
```

Comment:
I think the race we're describing is:
- Thread 1 deletes a file and reads `finished = false`
- Thread 2 finishes background migration. We set `finished = true`
- Thread 2 deletes all tombstones during live_restore cleanup
- Thread 1 creates a tombstone even though live restore is completed

We'll want a lock in that case. This ties into the latest conversation on slack about a `live_restore.state` file so I'm happy to leave this race with a FIXME and address it separately.

Reply:
a1c41de00